nixos/tests/make-test-hardware.nix

```diff
@@ -0,0 +1,78 @@
+/* Create tests that run in the nix sandbox but get access to the host hardware
```

Comment:
I actually would rename this to `make-impure-test.nix` since there might be other use cases not related to hardware that might be covered by this. I could imagine to have a sandbox-relaxed flag in the future to allow network access.